tests/TestHelpers/HttpRequestHelper.php

```diff
@@ -216,17 +217,20 @@ public static function sendRequest(
 				$client
 			);
 
-			if ($response->getStatusCode() >= 400 && $response->getStatusCode() !== self::HTTP_TOO_EARLY) {
+			if ($response->getStatusCode() >= 400 && $response->getStatusCode() !== self::HTTP_TOO_EARLY && $response->getStatusCode() !== self::HTTP_CONFLICT) {
 				$sendExceptionHappened = true;
 			}
 
 			if ($debugResponses) {
 				self::debugResponse($response);
 			}
 			$sendCount = $sendCount + 1;
-			$loopAgain = !$sendExceptionHappened && ($response->getStatusCode() === self::HTTP_TOO_EARLY) && ($sendCount <= $sendRetryLimit);
+			$loopAgain = !$sendExceptionHappened && ($response->getStatusCode() === self::HTTP_TOO_EARLY ||
+						($response->getStatusCode() === self::HTTP_CONFLICT &&
+						substr($url, 0, 44) === self::MAKE_FOLDER_API)) &&
```

Comment:
currently this function has 47 usages but we only need to loop this when we get `409` while making a folder. so i think we should check for it hear rather than sending a flag from functions to function. or i can change it.
```suggestion
						($response->getStatusCode() === self::HTTP_CONFLICT &&
						substr($url, 0, 44) === self::MAKE_FOLDER_API)) &&
```

Reply:
We think that there can be very short race-conditions when a user is first created and then a folder (or file?) is created/uploaded "immediately". A `409` conflict response can be returned. Presumably there is some backend processing still finishing for the user creation, and the quickly-incoming request "thinks" that the user does not exist and returns a `409`.

In "normal" `When` and `Then` steps we probably do not want to retry - if we get a `409` response then we want to send that back to the test code, and it will decide what to do (probably a `Then` step will fail that is expecting a `2xx` response code).

So we need a way to know that the request is for a `Given` step that is creating a folder, or uploading a file. Those are the times when this "transient" `409` might happen and be OK to retry.

If we can think of some way to "calculate" from the parameters passed to `sendRequest()` and know if it is a folder create or file upload from  `Given` step, then we can do that. Otherwise, we will have to pass a flag down to `sendRequest()`. Note: the flag can have default `false` and so only the functions that actually need to pass it through or set it `true` need to be modified.